### PR TITLE
EFR32 : Wifi Code movement from SDK support repo to CSA

### PR DESCRIPTION
**Pull request by sharad-patil24**
EFR32 
====

EFR32 Wifi implementation was present in SDK Support repo ( https://github.com/SiliconLabs/sdk_support ).
The Implementation is moved to ( src/platform/EFR32 ) folder to meet the standards.

### CODE REVIEW

File: src/platform/EFR32/wifi/dhcp_client.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/*****************************************************************************
+ * @file
```
Comment (andy31415) [inline src/platform/EFR32/wifi/dhcp_client.cpp:2]: If these files are to reside in the chip repo, I believe a license attribution/blurb update is required.

Comment (sharad-patil24) [reply src/platform/EFR32/wifi/dhcp_client.cpp]: Updated license attribution/blurb  for newly added files

File: examples/platform/efr32/rs911x/hal/rsi_board_configuration.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#ifndef _RSI_BOARD_CONFIGURATION_H_
```
Comment (andy31415) [inline examples/platform/efr32/rs911x/hal/rsi_board_configuration.h:20]: if you have pracma once, these ifndefs are not needed (applies to other files as well)

File: examples/platform/efr32/rs911x/hal/rsi_hal_mcu_interrupt.c [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * Includes
```
Comment (andy31415) [inline examples/platform/efr32/rs911x/hal/rsi_hal_mcu_interrupt.c:19]: nit: I would consider removing this comment - seems to use vertical space without helping me read the code easier ... I know `#include` are includes ...

**Review by andy31415 — APPROVED** [review]
I did not really review the code - leaving it up to the platform maintainers to do that.

File: examples/platform/efr32/rs911x/hal/rsi_board_configuration.h (hunk 2 above)
Comment (sharad-patil24) [reply examples/platform/efr32/rs911x/hal/rsi_board_configuration.h:18]: removed #pragma when #ifndef is used

File: examples/platform/efr32/rs911x/hal/rsi_hal_mcu_interrupt.c (hunk 3 above)
Comment (sharad-patil24) [reply examples/platform/efr32/rs911x/hal/rsi_hal_mcu_interrupt.c]: removed comment

**Pull request closed without merging**